### DISCUSSION TITLE
How does one use the latest sys.input functionality?

### DISCUSSION THREAD

**Discussion by justpankajdotcom** [Myriad-Dreamin/typst.ts#483] [Q&A] [answered]
Hi, thanks for supporting and maintaining this library.

I am having some trouble understanding the new developments going on with this project. Under the package directory, there are several packages starting from typst.ts, typst.react etc. What I understand is, typst.ts is the core package which has all the compiler implementation and the respective packages like typst.react and typst.vue is for facilitating rending in different frameworks. Is my understanding correct here?

Then there is another package which typst.node, only which has the latest changes like `sys.input` that typst has introduced. Is it like typst.node to be seen as replacement to typst.react and other packages? I am failing to understand why was the new package needed? And if possible, can you show a small snippet on how to use the `sys.input` functionality please? I am trying to replace values from user input inside the typst template. Thanks for the support. 

My dev dependency - 

```
"@myriaddreamin/typst-ts-web-compiler": "0.4.2-rc6"
  "@myriaddreamin/typst.ts": "0.4.2-rc6"
```

**Comment by Myriad-Dreamin** [comment, marked as answer]
Hello, `sys.input` will land in typst v0.11.0, and currently we are on typst v0.10.0. After the next typst update, I'll update the document for you along with new APIs.

**Reply by justpankajdotcom** [reply]
Hi, thank you so much for clarifying it.

**Reply by aszenz** [reply]
@Myriad-Dreamin I couldn't find anything about sys.input in the current docs, is this feature already exposed in the web api?

**Reply by Myriad-Dreamin** [reply]
@aszenz sorry I haven't added it to web api. There is a sys.inputs parameter in node api tho, thus this is only lack of exposing the parameter to web api as well. https://github.com/Myriad-Dreamin/typst.ts/blob/af4962758e6ed2e8bbfe4f4ad14086d078ccb0b2/packages/typst.node/src/lib.rs#L122-L123

**Reply by aszenz** [reply]
Cool, does this function need to be exposed: 
https://github.com/Myriad-Dreamin/typst.ts/blob/af4962758e6ed2e8bbfe4f4ad14086d078ccb0b2/packages/compiler/src/lib.rs#L151

**Reply by Myriad-Dreamin** [reply]
@aszenz sorry, I was really busy and happened to ignore the message in my email. Yes, this is the exact function to expose.

**Reply by Myriad-Dreamin** [reply]
https://github.com/Myriad-Dreamin/typst.ts/pull/595

**Comment by Myriad-Dreamin** [comment]
> What I understand is, typst.ts is the core package which has all the compiler implementation and the respective packages like typst.react and typst.vue is for facilitating rending in different frameworks. Is my understanding correct here?

It is correct.

> `typst.node`

It is a system-level [node binding](https://nodejs.org/api/n-api.html) to typst compiler, it doesn't compile into wasm but native machine code, and it cannot run in browsers. I was considering merge `typst.ts` and `typst.node`, but I found the compiler API with/without system file system differs a lot, so I splitted them. I'm not sure whether I will merge them in future.
